kusto/ingest/ingest.go

```diff
@@ -440,7 +441,87 @@ var (
 	ErrTooLarge = errors.ES(errors.OpIngestStream, errors.KClientArgs, "cannot add data larger than 4MiB")
 )
 
-const mib = 1024 * 1024
+const (
+	MaxRetryCount    = 3
+	mib              = 1024 * 1024
+	MaxStreamingSize = 4 * mib
+)
+
+// FromFile allows uploading a data file for Kusto from either a local path or a blobstore URI path.
+// This method is thread-safe.
+func (i *Ingestion) FromFileManaged(ctx context.Context, fPath string,
+	options ...FileOption) (*Result, error) {
+	result, props, err := i.prepForIngestion(ctx, options)
+	if err != nil {
+		return nil, err
+	}
+
+	result.record.IngestionSourcePath = fPath
+
+	local, err := filesystem.IsLocalPath(fPath)
+	if err != nil {
+		return nil, err
+	}
+
+	skipStreaming := false
+
+	if local {
```

Comment:
And if it is a huge blob?
Not sure if it can be provided, but do you check the ingestion properties for the file size?
If there's no such option in Go maybe we need to support it or at least request this value in the Managed APIs.

Reply:
If it's a blob it's automatically queued Ingestion, because downloading it only to send is inefficient.

If it's a local file we check it's size via Stat

Reply:
Where in the code is it redirected to queued?

Reply:
In the code inside if skipStreaming

Reply:
Now the skip streaming is correct